packages/core/src/utils/objectCheck.ts

```diff
@@ -0,0 +1,44 @@
+import type { ProofAttributeInfo } from '../modules/proofs'
+
+export class ObjectCheck {
+  public static objectEquals(x: Map<string, ProofAttributeInfo>, y: Map<string, ProofAttributeInfo>): boolean {
+    'use strict'
+    if (x === null || x === undefined || y === null || y === undefined) {
+      return x === y
+    }
+    // after this just checking type of one would be enough
+    if (x.constructor !== y.constructor) {
+      return false
+    }
+    // if they are functions, they should exactly refer to same one (because of closures)
+    if (x instanceof Function) {
+      return x === y
+    }
+    // if they are regexps, they should exactly refer to same one (it is hard to better equality check on current ES)
+    if (x instanceof RegExp) {
+      return x === y
+    }
+    if (x === y || x.valueOf() === y.valueOf()) {
+      return true
+    }
+
+    // if they are dates, they must had equal valueOf
+    if (x instanceof Date) {
+      return false
+    }
+
+    // if they are strictly equal, they both need to be object at least
+    if (!(x instanceof Object)) {
+      return false
+    }
+    if (!(y instanceof Object)) {
+      return false
+    }
+
+    // recursive object equality check
+    const p = Object.keys(x)
+    return Object.keys(y).every(function (i) {
+      return p.indexOf(i) !== -1
+    })
```

Comment:
This doesn't seem recursive to me?

Reply:
I think we should leave the comment and make it recursive instead as otherwise we're not doing a full equality check

Reply:
Hmm...ok well here we need to make a function recursive that takes two maps as arguments.

For that last statement I am kind of thinking something along the lines of:

```
const xkeys = Object.keys(x)
  const ykeys = Object.keys(y)
  return (
    ykeys.every(function (i) {
      return xkeys.indexOf(i) !== -1
    }) &&
    xkeys.every(function (i) {
      const a: Map<string, unknown> = new Map(Object.entries(i))
      const b: Map<string, unknown> = new Map(Object.entries(ykeys))
      return objectEquals(a, b)
    })
  )
```

Reply:
I have created a rudimentary recursive version and written a unit test to test object (map) equality.

Probably a neater way of doing the same thing.